weather_mv/loader_pipeline/sinks.py

```diff
@@ -72,15 +73,71 @@ def _make_grib_dataset_inmem(grib_ds: xr.Dataset) -> xr.Dataset:
     return data_ds
 
 
-def _preprocess_tif(ds: xr.Dataset, filename: str, tif_metadata_for_datetime: str) -> xr.Dataset:
+def match_datetime(file_name: str, regex_expression: str) -> datetime.datetime:
+    """Matches the regex string given and extracts the datetime object.
+
+    Args:
+        file_name: File name from which you want to extract datetime.
+        regex_expression: Regex expression for extracting datetime from the filename.
+
+    Returns:
+        A datetime object after extracting from the filename.
+    """
+    def get_time_order(time_string):
+        idx = [i for i in range(len(time_string)) if time_string.startswith('%', i)]
+        idx = [i+1 for i in idx]
+        order_list = ['%'+time_string[i] for i in idx]
+        return order_list
+
+    def rearrange_time_list(order_list, time_list):
+        order_reference_list = ['%Y', '%m', '%d', '%H', '%M', '%S']
+        if order_list == order_reference_list:
+            return time_list
+        new_time_list = []
+        print(order_list)
+        for i, j in zip(order_list, time_list):
+            dst = order_reference_list.index(i)
+            new_time_list.insert(dst, j)
+        return new_time_list
+    char_to_replace = {
+        '%Y': ['([0-9]{4})', [0, 1978]],
+        '%m': ['([0-9]{2})', [1, 1]],
+        '%d': ['([0-9]{2})', [2, 1]],
+        '%H': ['([0-9]{2})', [3, 0]],
+        '%M': ['([0-9]{2})', [4, 0]],
+        '%S': ['([0-9]{2})', [5, 0]],
+        '*': ['.*']
+    }
```

Comment:
Idea for the parser: let's use a combination of `fnmatch` [1] and datetime's standard parsing. 

[1]: https://docs.python.org/3/library/fnmatch.html

I'd still prefer to use standard library logic instead of implementing something on our own -- especially when a regex is involved :)

Reply:
Resolved in a 1:1 meeting.